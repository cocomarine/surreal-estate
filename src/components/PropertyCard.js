import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faBath,
  faBed,
  faSterlingSign,
} from "@fortawesome/free-solid-svg-icons";
import { faEnvelope } from "@fortawesome/free-regular-svg-icons";
import { faFortAwesome } from "@fortawesome/free-brands-svg-icons";
import "../styles/property-card.css";

const PropertyCard = ({ fields }) => {
  const { title, type, bathrooms, bedrooms, price, city, email } = fields;

  return (
    <div className="property-card-container">
      <FontAwesomeIcon icon={faFortAwesome} size="2x" className="castle-icon" />
      <div className="property-card__title">{title}</div>
      <div className="property-card__type__city">
        {type} - {city}
      </div>
      <div className="property-card__bathrooms">
        <FontAwesomeIcon
          icon={faBed}
          className="bedIcon"
          data-testid="bed-icon"
        />
        {bathrooms}
      </div>
      <div className="property-card__bedrooms">
        <FontAwesomeIcon
          icon={faBath}
          className="bathIcon"
          data-testid="bath-icon"
        />
        {bedrooms}
      </div>
      <div className="property-card__price">
        <FontAwesomeIcon
          icon={faSterlingSign}
          className="sterlingIcon"
          data-testid="sterling-icon"
        />
        {price}
      </div>
      <a href={`mailto:${email}`} className="property-card__email">
        <button type="submit" className="email-button">
          <FontAwesomeIcon
            icon={faEnvelope}
            className="emailIcon"
            data-testid="email-icon"
          />
          Email
        </button>
      </a>
    </div>
  );
};

PropertyCard.propTypes = {
  fields: PropTypes.shape({
    title: PropTypes.string,
    city: PropTypes.string,
    type: PropTypes.string,
    bedrooms: PropTypes.string,
    bathrooms: PropTypes.string,
    price: PropTypes.string,
    email: PropTypes.string,
  }).isRequired,
};

export default PropertyCard;
