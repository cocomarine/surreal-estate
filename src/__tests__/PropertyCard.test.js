import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "test title",
    city: "test city",
    type: "test type",
    bedrooms: "2",
    bathrooms: "1",
    price: "1234",
    email: "test@email.com",
  };

  it("renders correctly", () => {
    const rendered = renderer.create(<PropertyCard fields={validProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    render(<PropertyCard fields={validProps} />);

    expect(screen.getByText("test title")).toHaveClass("property-card__title");
    expect(screen.getByText("test type - test city")).toHaveClass(
      "property-card__type__city"
    );
    expect(screen.getByText("2")).toHaveClass("property-card__bedrooms");
    expect(screen.getByText("1")).toHaveClass("property-card__bathrooms");
    expect(screen.getByText("1234")).toHaveClass("property-card__price");
    expect(screen.getByText("Email")).toHaveClass("email-button");

    expect(screen.getByTestId("bed-icon")).toHaveClass("bedIcon");
    expect(screen.getByTestId("bath-icon")).toHaveClass("bathIcon");
    expect(screen.getByTestId("email-icon")).toHaveClass("emailIcon");
  });
});
