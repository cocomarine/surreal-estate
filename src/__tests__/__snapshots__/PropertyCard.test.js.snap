// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<div
  className="property-card-container"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-fort-awesome fa-2x castle-icon"
    data-icon="fort-awesome"
    data-prefix="fab"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 512 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M489.2 287.9h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6V146.2c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32h-36.6v-32c0-6-8-4.6-11.7-4.6v-38c8.3-2 17.1-3.4 25.7-3.4 10.9 0 20.9 4.3 31.4 4.3 4.6 0 27.7-1.1 27.7-8v-60c0-2.6-2-4.6-4.6-4.6-5.1 0-15.1 4.3-24 4.3-9.7 0-20.9-4.3-32.6-4.3-8 0-16 1.1-23.7 2.9v-4.9c5.4-2.6 9.1-8.3 9.1-14.3 0-20.7-31.4-20.8-31.4 0 0 6 3.7 11.7 9.1 14.3v111.7c-3.7 0-11.7-1.4-11.7 4.6v32h-36.6v-32c0-2.6-2-4.6-4.6-4.6h-27.4c-2.6 0-4.6 2-4.6 4.6v32H128v-32c0-2.6-2-4.6-4.6-4.6H96c-2.6 0-4.6 2-4.6 4.6v178.3H54.8v-32c0-2.6-2-4.6-4.6-4.6H22.8c-2.6 0-4.6 2-4.6 4.6V512h182.9v-96c0-72.6 109.7-72.6 109.7 0v96h182.9V292.5c.1-2.6-1.9-4.6-4.5-4.6zm-288.1-4.5c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64zm146.4 0c0 2.6-2 4.6-4.6 4.6h-27.4c-2.6 0-4.6-2-4.6-4.6v-64c0-2.6 2-4.6 4.6-4.6h27.4c2.6 0 4.6 2 4.6 4.6v64z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  <div
    className="property-card__title"
  >
    test title
  </div>
  <div
    className="property-card__type__city"
  >
    test type
     - 
    test city
  </div>
  <div
    className="property-card__bathrooms"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bed bedIcon"
      data-icon="bed"
      data-prefix="fas"
      data-testid="bed-icon"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 640 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M32 32c17.7 0 32 14.3 32 32V320H288V160c0-17.7 14.3-32 32-32H544c53 0 96 43 96 96V448c0 17.7-14.3 32-32 32s-32-14.3-32-32V416H352 320 64v32c0 17.7-14.3 32-32 32s-32-14.3-32-32V64C0 46.3 14.3 32 32 32zm144 96a80 80 0 1 1 0 160 80 80 0 1 1 0-160z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    1
  </div>
  <div
    className="property-card__bedrooms"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bath bathIcon"
      data-icon="bath"
      data-prefix="fas"
      data-testid="bath-icon"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M96 77.3c0-7.3 5.9-13.3 13.3-13.3c3.5 0 6.9 1.4 9.4 3.9l14.9 14.9C130 91.8 128 101.7 128 112c0 19.9 7.2 38 19.2 52c-5.3 9.2-4 21.1 3.8 29c9.4 9.4 24.6 9.4 33.9 0L289 89c9.4-9.4 9.4-24.6 0-33.9c-7.8-7.9-19.8-9.1-29-3.8C246 39.2 227.9 32 208 32c-10.3 0-20.2 2-29.2 5.5L163.9 22.6C149.4 8.1 129.7 0 109.3 0C66.6 0 32 34.6 32 77.3V256c-17.7 0-32 14.3-32 32s14.3 32 32 32H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H96V77.3zM32 352v16c0 28.4 12.4 54 32 71.6V480c0 17.7 14.3 32 32 32s32-14.3 32-32V464H384v16c0 17.7 14.3 32 32 32s32-14.3 32-32V439.6c19.6-17.6 32-43.1 32-71.6V352H32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    2
  </div>
  <div
    className="property-card__price"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-sterling-sign sterlingIcon"
      data-icon="sterling-sign"
      data-prefix="fas"
      data-testid="sterling-icon"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M112 160.4c0-35.5 28.8-64.4 64.4-64.4c6.9 0 13.8 1.1 20.4 3.3l81.2 27.1c16.8 5.6 34.9-3.5 40.5-20.2s-3.5-34.9-20.2-40.5L217 38.6c-13.1-4.4-26.8-6.6-40.6-6.6C105.5 32 48 89.5 48 160.4V224H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H48v44.5c0 17.4-4.7 34.5-13.7 49.4L4.6 431.5c-5.9 9.9-6.1 22.2-.4 32.2S20.5 480 32 480H288c17.7 0 32-14.3 32-32s-14.3-32-32-32H88.5l.7-1.1C104.1 390 112 361.5 112 332.5V288H224c17.7 0 32-14.3 32-32s-14.3-32-32-32H112V160.4z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    1234
  </div>
  <a
    className="property-card__email"
    href="mailto:test@email.com"
  >
    <button
      className="email-button"
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-envelope emailIcon"
        data-icon="envelope"
        data-prefix="far"
        data-testid="email-icon"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64 112c-8.8 0-16 7.2-16 16v22.1L220.5 291.7c20.7 17 50.4 17 71.1 0L464 150.1V128c0-8.8-7.2-16-16-16H64zM48 212.2V384c0 8.8 7.2 16 16 16H448c8.8 0 16-7.2 16-16V212.2L322 328.8c-38.4 31.5-93.7 31.5-132 0L48 212.2zM0 128C0 92.7 28.7 64 64 64H448c35.3 0 64 28.7 64 64V384c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      Email
    </button>
  </a>
</div>
`;
